components/discount/renewalDiscount.tsx

```diff
@@ -103,29 +101,24 @@ const RenewalDiscount: FunctionComponent<RenewalDiscountProps> = ({
   const router = useRouter();
   const needsAllowance = useAllowanceCheck(displayedCurrency, address);
   const tokenBalances = useBalances(address); // fetch the user balances for all whitelisted tokens
+  const [hasChoseCurrency, setHasChoseCurrency] = useState<boolean>(false);
   const [loadingPrice, setLoadingPrice] = useState<boolean>(false);
 
   useEffect(() => {
-    if (!renewData?.transaction_hash || !salts || !metadataHashes) return;
+    if (!renewData?.transaction_hash || !salt || !metadataHash) return;
     // register the metadata to the sales manager db
-    Promise.all(
-      salts.map((salt, index) =>
-        fetch(`${process.env.NEXT_PUBLIC_SALES_SERVER_LINK}/add_metadata`, {
-          method: "POST",
-          headers: { "Content-Type": "application/json" },
-          body: JSON.stringify({
-            meta_hash: metadataHashes[index],
-            email,
-            tax_state: isSwissResident ? "switzerland" : "none",
-            salt: salt,
-          }),
-        })
-      )
-    )
-      .then((responses) => Promise.all(responses.map((res) => res.json())))
-      .catch((error) => {
-        console.log("Error on sending metadata:", error);
-      });
+    fetch(`${process.env.NEXT_PUBLIC_SALES_SERVER_LINK}/add_metadata`, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        meta_hash: metadataHash,
+        email: email,
+        tax_state: isSwissResident ? "switzerland" : "none",
+        salt: salt,
+      }),
+    })
+      .then((res) => res.json())
+      .catch((err) => console.log("Error on sending metadata:", err));
```

Comment:
Why do you pass this as something with only one hash, I don't understand the link with smart currency choosing ?

Reply:
It's an update we made for it to work with the subscriptions, now a user has only one `metadatahash`. And actually we should only store it if `needMetadata = true` so I will update it.